### ISSUE TITLE
Footnote formatting issue

### ISSUE THREAD

**Issue by samarulmeu** [previm/previm#207]
When the footnote starts with a `*` (used for italic or bold) *previm* fails with the following error

```
Error detected while processing BufWritePost Autocommands for "<buffer=1>"..function previm#refresh[6]..previm#refresh_js_function[1]..<SNR>110_function_template[2]..previm#convert_to_content[16]..previm#convert_relative_to_absolute_filepath[1]..previm#relative_to_absolute_filepath:
line   59:
E871: (NFA regexp) Can't have a multi follow a multi
```

You can use this markdown file example to reproduce the error.

```
# Testing footnote

Lorem ipsum[^1] dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.

[^1]: *Footnote*
```

**Comment by lambdalisue** [comment]
Could you try #209?

**Comment by samarulmeu** [comment]
@lambdalisue I think you've meant #208

**Comment by KSR-Yasuda** [comment]
Perhaps doesn't this work?

```diff
--- a/autoload/previm.vim
+++ b/autoload/previm.vim
@@ -324,7 +324,7 @@ function! previm#convert_to_content(lines) abort
 endfunction
 
 function! previm#convert_relative_to_absolute_filepath(text, mkd_dir) abort
-  return substitute(a:text, '\v!?\[%([^\[\]\(\)]+|\[[^\[\]]*\]|\([^()]*\))*\]\([^()]*\)|^\[([^\[\]]*)\]:\s*(.*)', '\=previm#relative_to_absolute_filepath(submatch(0), a:mkd_dir)', 'g')
+  return substitute(a:text, '\v!?\[%([^\[\]\(\)]+|\[[^\[\]]*\]|\([^()]*\))*\]\([^()]*\)|^\[[^^]%([^\[\]]*)\]:\s*%(.*)', '\=previm#relative_to_absolute_filepath(submatch(0), a:mkd_dir)', 'g')
 endfunction
 
 " convert example
```

This matching regexp has two parts:

1. `!?\[%([^\[\]\(\)]+|\[[^\[\]]*\]|\([^()]*\))*\]\([^()]*\)`
    - Inline link / image styles (i.e., `[text](URL)` or `![text](URL)`)
2. `^\[([^\[\]]*)\]:\s*(.*)`
    - Reference styles (i.e., `[label]: URL` (at the line beginning))

```text
\v!?\[%([^\[\]\(\)]+|\[[^\[\]]*\]|\([^()]*\))*\]\([^()]*\)|^\[([^\[\]]*)\]:\s*(.*)
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ^^^^^^^^^^^^^^^^^^^^^^^
  1. Inline style part                                     2. Reference style part
```

Footnote syntax (`[^footnote-label]: Footnote text`) matches the latter.

To avoid the URL conversion for the footnote syntax, make the latter pattern not to match `^`-beginning labels.

```text
Old: ^\[                   ([^\[\]]*)            \]:\s*      (.*)
                           ^^^^^^^^^^                        ^^^^
                           Link reference label              URL

New: ^\[ [^^]             %([^\[\]]*)            \]:\s*     %(.*)
         ^^^^             ^^^^^^^^^^^                       ^^^^^
         Non-hat char      Link reference label              URL
                           (Added `%` for the paren,         (Also added `%`)
                           that part is
                           not used by submatch ref)
```

This change requires the latter reference-style labels to be at least one character,
but Markdown syntax originally would do so (since they must be some recognizable label text), wouldn't it?

**Comment by lambdalisue** [comment]
Should be closed via #214

**Comment by samarulmeu** [comment]
With the latest commit all seems to be OK.
